src/util/records.py

```diff
@@ -106,6 +107,18 @@ def remove_view(cr, xml_id=None, view_id=None, silent=False, key=None):
         for [v_id] in cr.fetchall():
             remove_view(cr, view_id=v_id, silent=silent, key=xml_id)
 
+    if not key and column_exists(cr, "ir_ui_view", "key"):
+        cr.execute("SELECT key FROM ir_ui_view WHERE id = %s and key != %s", [view_id, xml_id])
+        [key] = cr.fetchone() or [None]
+
+    # Occurrences of xml_id and key in the t-call of views are to be found and removed.
+    if xml_id != "?" or key:
+        if xml_id != key:
+            remove_redudant_tcalls(cr, xml_id)
+            remove_redudant_tcalls(cr, key)
+        else:
+            remove_redudant_tcalls(cr, xml_id)
```

Comment:
Can be made more direct. Also if xml_id is ? you don't need to remove by xml_id.
```suggestion
    if xml_id != "?":
        remove_redundant_tcalls(cr, xml_id)
    if key and key != xml_id:
        remove_redundant_tcalls(cr, key)        
```